schemas/v3.1/schema.yaml

```
@@ -105,11 +105,11 @@ $defs:
         format: uri
     required:
       - name
-    oneOf:
-      - required:
-        - identifier
-      - required:
-        - url
+    dependentSchemas:
+      identifier:
+        not:
+          required:
+            - url
```

Doesn't it go the other way as well? if `url` exists, then we can't have `identifier`.

```suggestion
    dependentSchemas:
      identifier:
        not:
          required:
            - url
      url:
        not:
          required:
            - identifier            
```

---

This could also be done more succinctly with just `maxProperties: 2`, since `name` is required and x- properties aren't permitted here.

---

It's not necessary. What you propose was one of my initial options, but @jdesrosiers proposed that shorter variation:
> I didn't include the inverse because it isn't necessary and produces unnecessary error results.  

See [discussion here](https://github.com/OAI/OpenAPI-Specification/issues/2975#issuecomment-1192824618)